exporter/chronicleexporter/exporter.go

```diff
@@ -61,17 +67,38 @@ func newExporter(cfg *Config, params exporter.CreateSettings) (*chronicleExporte
 		return nil, fmt.Errorf("dial: %w", err)
 	}
 
-	marshaller, err := newProtoMarshaler(*cfg, params.TelemetrySettings, buildLabels(cfg))
+	customerID, err := uuid.Parse(cfg.CustomerID)
+	if err != nil {
+		return nil, fmt.Errorf("parse customer ID: %w", err)
+	}
+
+	marshaller, err := newProtoMarshaler(*cfg, params.TelemetrySettings, buildLabels(cfg), customerID[:])
 	if err != nil {
 		return nil, fmt.Errorf("create proto marshaller: %w", err)
 	}
 
-	return &chronicleExporter{
-		cfg:       cfg,
-		logger:    params.Logger,
-		client:    api.NewIngestionServiceV2Client(conn),
-		marshaler: marshaller,
-	}, nil
+	uuidCID, err := uuid.Parse(collectorID)
+	if err != nil {
+		return nil, fmt.Errorf("parse collector ID: %w", err)
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	exp := &chronicleExporter{
+		cfg:         cfg,
+		logger:      params.Logger,
+		metrics:     newExporterMetrics(uuidCID[:], customerID[:], exporterID, cfg.Namespace),
+		client:      api.NewIngestionServiceV2Client(conn),
+		marshaler:   marshaller,
+		collectorID: collectorID,
+		exporterID:  exporterID,
+		cancel:      cancel,
+	}
+
+	if cfg.CollectAgentMetrics {
+		go exp.startHostMetricsCollection(ctx)
+	}
```

Comment:
We aren't tracking this goroutine at all. So when `Shutdown` is called there is no guarantee this will be finished by the time it exits. We should use a channel or wait group to signal this has exited.

Reply:
Added.